File: 00._Course_Material/01._Assignments/01._Introduction_Data_Formats/01a._solution/Node-Parsing/modules/csv_parser.js
const fs = require("fs");
const { parse } = require("csv-parse");

class CSVParser {
  constructor() {
    this.data = [];
  }

  // consdering using async/await or new Promise() to handle the async nature of the file read
  parse(filePath) {
    fs.createReadStream(filePath)
    .pipe(parse({ delimiter: "," }))
    .on("data", (row) => {
        this.data.push(row);
    })
    .on("end",() =>{
        console.log("CSV Data:", this.data);
    })
    .on("error", (err) => {
        console.error("Error parsing CSV file:", error);
    });
  }

  get() {
    return this.data;
  }
}

export default CSVParser

File: 00._Course_Material/01._Assignments/01._Introduction_Data_Formats/01a._solution/Python-Parsing/modules/csv_parser.py
import csv


# Parse CSV
def parse_csv(file):
    with open(file, newline='', mode='r') as f:
        reader = csv.DictReader(f)
        data = [row for row in reader]
    print(f"CSV Data from {file}:", data)
    return { "data": data }